diff --git a/node_modules/superagent/lib/node/index.js b/node_modules/superagent/lib/node/index.js
index 2f948a9..8ae8de1 100644
--- a/node_modules/superagent/lib/node/index.js
+++ b/node_modules/superagent/lib/node/index.js
@@ -716,6 +716,21 @@ Request.prototype.request = function(){
   // initiate request
   const mod = this._enableHttp2 ? exports.protocols['http2:'].setProtocol(url.protocol) : exports.protocols[url.protocol];
 
+  // use proxy
+  console.log('proxy environment', process.env.http_proxy)
+  if(process.env.http_proxy) {
+    var proxy = parse(process.env.http_proxy);
+    console.log('http proxy', proxy)
+    if (url.protocol === 'http:') {
+      options.port = proxy.port;
+      options.path = this.url;
+      options.host = proxy.hostname;
+      console.log('http proxy created')
+    } else if (url.protocol === 'https:') {
+      console.log('do not support https proxy')
+    }
+  }
+
   // request
   const req = (this.req = mod.request(options));
 
